### PR TITLE
Fixshowbgp

### PR DESCRIPTION
**Pull request by ee-github**
## Description
Fixed the 'show bgp vrf all all summary" NXOS parser to handle output where neighbors have a 4byte ASN.

## Motivation and Context
The impact of the bug is that certain neighbors are ignored in genie test cases.  There is no impact from this bugfix other than now including those neighbors.

## Impact (If any)
Currently when a neighbor has a 4byte ASN the CLI output table spills that neighbor's info across two lines as in the example below.  This causes the parser to completely exclude that neighbor from the output.

Sanitized example from a Nexus 3548XL running 9.3.7:

```
SWITCH01# sh bgp vrf all all summary 
BGP summary information for VRF default, address family IPv4 Unicast
BGP router identifier 1.1.1.1, local AS number 64543
BGP table version is 3272, IPv4 Unicast config peers 8, capable peers 6
510 network entries and 1025 paths using 186240 bytes of memory
BGP attribute entries [96/16512], BGP AS path entries [34/396]
BGP community entries [15/556], BGP clusterlist entries [0/0]

Neighbor        V    AS MsgRcvd MsgSent   TblVer  InQ OutQ Up/Down  State/PfxRcd
10.10.10.1      4 64523  577557  577451     3272    0    0     9w3d 109       
10.10.20.1      4 64544       0       0        0    0    0     9w3d Shut (Admin)
10.10.30.1      4 64544       0       0        0    0    0     9w3d Shut (Admin)
10.10.40.1      4 64544   96328   96376     3272    0    0    2d17h 9         
10.10.50.1      4 64544   96317   96355     3272    0    0    2d17h 9         
10.10.60.1      4 64543   96816   96387     3272    0    0     9w3d 504       
10.10.70.1      4 64597  578074  577468     3272    0    0     9w3d 89        
10.10.80.1      4 4210410100
                         578124  577455     3272    0    0     9w3d 297
```

## Screenshots:
```
genie parse 'show bgp vrf all all summary' --testbed-file testbed.yaml --devices SWITCH01 | jq '.vrf.default.neighbor|keys'
[
  "10.10.10.1",
  "10.10.20.1",
  "10.10.30.1",
  "10.10.40.1",
  "10.10.50.1",
  "10.10.60.1",
  "10.10.70.1"
]
```

After:

```
genie parse 'show bgp vrf all all summary' --testbed-file testbed.yaml --devices SWITCH01 | jq '.vrf.default.neighbor|keys'
[
  "10.10.10.1",
  "10.10.20.1",
  "10.10.30.1",
  "10.10.40.1",
  "10.10.50.1",
  "10.10.60.1",
  "10.10.70.1",
  "10.10.80.1"
]
```

## Checklist:

- [x] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

**Comment by ee-github** [comment]
@Taarini I believe I can.  After poking around it looks like NXOS now follows the "folder" testing scheme.  I should be able to add another test case there that covers the conditions for this PR by following one of the existing tests as an example.  Give me a bit of time.  (Already tested on a live device)

**Review by Taarini — COMMENTED** [review]
sure

**Comment by ee-github** [comment]
@Taarini alright hopefully that does the trick.  I added a 4byte ASN test case.

File: changelog/undistributed/changelog_show_bgp_vrf_all_all_summary_nxos_20220628.rst
Comment (Taarini) [inline changelog/undistributed/changelog_show_bgp_vrf_all_all_summary_nxos_20220628.rst:7]: Can you please edit the timestamp in filename to changelog_show_bgp_vrf_all_all_summary_nxos_%Y%M%D%H%M%S.rst

File: src/genie/libs/parser/nxos/show_bgp.py
Comment (Taarini) [inline src/genie/libs/parser/nxos/show_bgp.py:3606]: Can you please add comment like below,
```suggestion
        # Neighbor        V             AS MsgRcvd MsgSent   TblVer  InQ OutQ Up/Down  State/PfxRcd
        # 10.10.10.10     4 4211111111
        p8_3 = re.compile(r'^\s*(?P<neighbor>[a-zA-Z0-9\.\:]+) +(?P<v>[0-9]+) +(?P<as>[0-9]+)$')
        # Neighbor        V    AS MsgRcvd MsgSent   TblVer  InQ OutQ Up/Down  State/PfxRcd
        #                                                  0              0           0      0        0       5w6d   Idle 
        p8_4 = re.compile(r'^\s*(?P<msg_rcvd>[0-9]+) +(?P<msg_sent>[0-9]+)'
                          r' +(?P<tbl_ver>[0-9]+) +(?P<inq>[0-9]+)'
                          r' +(?P<outq>[0-9]+) +(?P<up_down>[a-zA-Z0-9\:]+)'
                          r' +(?P<state_pfxrcd>(?P<state>[a-zA-Z\s\(\)]+)?(?P<prx_rcd>\d+)?([\w\(\)\s]+)?)$')                            
```

**Review by Taarini — CHANGES_REQUESTED** [review]

File: changelog/undistributed/changelog_show_bgp_vrf_all_all_summary_nxos_20220628.rst
Comment (ee-github) [reply changelog/undistributed/changelog_show_bgp_vrf_all_all_summary_nxos_20220628.rst]: Done.  Pushed another commit with file renamed.

**Review by Taarini — CHANGES_REQUESTED** [review]
Can you check why tests are failing

**Comment by ee-github** [comment]
@Taarini would you please kick off the tests again?  I don't believe I can do it because "First-time contributors need a maintainer to approve running workflows".

**Comment by Taarini** [comment]
> 

What is the version of pyats being used by you

**Comment by ee-github** [comment]
I am using 22.5

**Comment by Taarini** [comment]
> I am using 22.5

Can you please upgrade to 22.6. Since pyats 22.6 was released yesterday.

**Comment by ee-github** [comment]
As you maybe can tell I'm not super git savvy.  I pull latest master into my branch and then pushed again.  Hopefully that was correct and not includes 22.6 files.  I did not see any conflict.

**Comment by ee-github** [comment]
Hi All.  Just checking in to see if any other action or info is required from my end.  Not sure on what cadence the team here reviews / merges contributions.  Thanks.

**Comment by Taarini** [comment]
You just have to merge the latest code with the master, if there are any recent push and check if the pipeline passes.

As of now, please wait for the other reviewers to review, after which if everything looks good your PR will be merged.

File: src/genie/libs/parser/nxos/show_bgp.py
Comment (domachad) [reply src/genie/libs/parser/nxos/show_bgp.py:3914]: I don't think this approach should be used as it ignores any issues and may be masking a bug. Moreover, if one of the keys in the middle is missing, the rest of them will not be added. I'd say assign them without the try/except block if all of them are mandatory. If some are optional, then check if they exist before adding them to the dictionary

Comment (domachad) [reply src/genie/libs/parser/nxos/show_bgp.py:3914]: sure, if this is done in other parts of the code as well, then it will have to be refactored. It can be done in a different pr.

**Review by domachad — APPROVED** [review]

**Pull request merged**
